backend/spec/controllers/spree/admin/payment_methods_controller_spec.rb

```diff
@@ -70,5 +70,33 @@ class GatewayWithPassword < PaymentMethod
         expect(assigns(:payment_methods).to_a).to eql([second_method, first_method])
       end
     end
+
+    describe "#update" do
+      # Regression test for https://github.com/solidusio/solidus/issues/2789
+      let(:payment_method) { create(:store_credit_payment_method) }
+      let(:params) do
+        {
+          id: payment_method.id,
+          payment_method: {
+            name: 'Check',
+            type: 'Spree::PaymentMethod::Check'
+          }
+        }
+      end
+
+      subject { put :update, params: params }
```

Comment:
@BenMorganIO because of the error I didn't use the `payment_method` reference, for example, here:

```
describe "#update" do
      # Regression test for https://github.com/solidusio/solidus/issues/2789
      let(:payment_method) { create(:store_credit_payment_method) }
      let(:params) do
        {
          id: payment_method.id,
          payment_method: {
            name: 'Check',
            type: 'Spree::PaymentMethod::Check'
          }
        }
      end

      it 'redirects to payment method edit page' do
        put :update, params: params
        expect(response).to redirect_to(spree.edit_admin_payment_method_path(payment_method))
      end

      it 'updates the resource' do
        expect { put :update, params: params }.to change { payment_method.reload.name }.from('Store Credit').to('Check')
      end
    end
```

I'm getting this:
```
ActiveRecord::RecordNotFound: Couldn't find Spree::PaymentMethod::StoreCredit with 'id'=1 [WHERE "spree_payment_methods"."type" IN ('Spree::PaymentMethod::StoreCredit')]
```

Reply:
Just need to instantiate payment_method before the specs. Since it's used by both tests, you can do:

```ruby
let!(:payment_method) { ... }
```

Reply:
doing that I got another error
```
1) Spree::Admin::PaymentMethodsController#index respects the order of payment methods by position
     Failure/Error: expect(assigns(:payment_methods).to_a).to eql([second_method, first_method])

       expected: [#<Spree::GatewayWithPassword id: 3, type: "Spree::GatewayWithPassword", name: "Second", description:...=>"1235"}, preference_source: nil, position: 2, available_to_users: true, available_to_admin: true>]
            got: [#<Spree::GatewayWithPassword id: 3, type: "Spree::GatewayWithPassword", name: "Second", description:...=>"1235"}, preference_source: nil, position: 3, available_to_users: true, available_to_admin: true>]

       (compared using eql?)

       Diff:
       @@ -1,3 +1,4 @@
        [#<Spree::GatewayWithPassword id: 3, type: "Spree::GatewayWithPassword", name: "Second", description: nil, active: true, deleted_at: nil, created_at: "2018-07-03 19:13:28", updated_at: "2018-07-03 19:13:28", auto_capture: nil, preferences: {:server=>"test", :test_mode=>true, :password=>"1235"}, preference_source: nil, position: 1, available_to_users: true, available_to_admin: true>,
       - #<Spree::GatewayWithPassword id: 2, type: "Spree::GatewayWithPassword", name: "First", description: nil, active: true, deleted_at: nil, created_at: "2018-07-03 19:13:28", updated_at: "2018-07-03 19:13:28", auto_capture: nil, preferences: {:server=>"test", :test_mode=>true, :password=>"1235"}, preference_source: nil, position: 2, available_to_users: true, available_to_admin: true>]
       + #<Spree::GatewayWithPassword id: 1, type: "Spree::GatewayWithPassword", name: "Bogus", description: nil, active: true, deleted_at: nil, created_at: "2018-07-03 19:13:28", updated_at: "2018-07-03 19:13:28", auto_capture: nil, preferences: {:server=>"test", :test_mode=>true, :password=>"haxme"}, preference_source: nil, position: 2, available_to_users: true, available_to_admin: true>,
       + #<Spree::GatewayWithPassword id: 2, type: "Spree::GatewayWithPassword", name: "First", description: nil, active: true, deleted_at: nil, created_at: "2018-07-03 19:13:28", updated_at: "2018-07-03 19:13:28", auto_capture: nil, preferences: {:server=>"test", :test_mode=>true, :password=>"1235"}, preference_source: nil, position: 3, available_to_users: true, available_to_admin: true>]
```